core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataIndexAliasesService.java

```diff
@@ -88,12 +88,16 @@ protected ClusterStateUpdateResponse newResponse(boolean acknowledged) {
 
                 @Override
                 public ClusterState execute(ClusterState currentState) {
-                    return innerExecute(currentState, request.actions());
+                    return updateAliasesInClusterState(currentState, request.actions());
                 }
             });
     }
 
-    ClusterState innerExecute(ClusterState currentState, Iterable<AliasAction> actions) {
+    /**
+     * Updates the given cluster state with the requested {@link AliasAction}s to perform
+     * and returns a new cluster state.
+     */
+    public final ClusterState updateAliasesInClusterState(ClusterState currentState, Iterable<AliasAction> actions) {
```

Comment:
One last question, why does it need to be visible for tests outside of this package? That is, why isn't it sufficient to construct a metadata alias service, submit a request to update the aliases, and then ask the cluster service for the cluster state?